service/src/main/java/org/apache/griffin/core/measure/entity/Rule.java

```diff
@@ -81,6 +81,9 @@ Licensed to the Apache Software Foundation (ASF) under one
     @Column(name = "\"out\"")
     private String out;
 
+    @JsonInclude(JsonInclude.Include.NON_NULL)
```

Comment:
It look like some other properties are implicit too -- in/out dataframe name, out, details. So making cache optional felt to be consistent with everything else.
As far as I undertsand, format-wise it's optional, and keeping it implicit will make API GET result consistent with body used to create measure object (if it was not provided, it won't appear in response; if it was provided, it will appear).
Another thing is if it will be explicit, all measures created before this field was introduced will have "null" in their body (or we'll need to set it to false by default, making GET response inconsistent with original content).

Assuming all said above, should it really be explicit field?

Reply:
do we have a standard api definition, implicit or explicit depend on what api style exposes to users

Reply:
@toyboxman don't see anything, service/api-guide.md does not describe any style guides or standard contracts, only shows examples. Also "cache" field was not documented in measure-configuration-guide.md, added information about it there. Javadoc says:
```
  * @param cache      cache the result for multiple usage (optional, valid for "spark-sql" and "df-ops" mode)
```
Taking into account it's valid not for all types of rules, hiding null values seems to be only valid behavior.

Reply:
@bhlx3lyx7 should we make this optional from service side?

Reply:
This parameter `cache` in rule configuration indicates spark will cache the output data frame, to avoid redundant calculation of the same data frame if it is used twice or more. 
In "griffin-dsl" mode, the rule will be translated internally, and the `cache` parameter would be filled by the translator. That's why it only works in "spark-sql" and "df-ops" mode.
In the current versions, UI doesn't support user defined "spark-sql" rules or "df-ops" rules, the `cache` parameter seems confusing for users either. But if we want to support user defined "spark-sql" rules, no matter through UI or service, it would be better to make the parameters configurable. It's ok to set it optional, I prefer to set it `false` by default.

Reply:
I think the default value is false by current implementation.